telemetry/definitions/commonDefinitions.json

```diff
@@ -567,20 +582,55 @@
             "description": "Open a window to view the Cloudfront Invalidation requests",
             "metadata": [{ "type": "result" }]
         },
+        {
+            "name": "cloudwatchlogs_copyArn",
+            "description": "Copy the ARN of a CloudWatch Logs entity",
+            "metadata": [
+                { "type": "result" },
+                { "type": "cloudWatchResourceType" }
+            ]
+        },
+        {
+            "name": "cloudwatchlogs_open",
+            "description": "Open a CloudWatch Logs entity. ServiceType indicates where the request came from (example: while viewing an ECS container)",
+            "metadata": [
+                { "type": "result" },
+                { "type": "cloudWatchResourceType" },
+                { "type": "serviceType", "required": false },
```

Comment:
Is `serviceType` optional because it's not always known prior to executing the user gesture, or is it optional because its absence implies that the user initiated the action directly on a CW resource?

These kinds of subtleties will become increasingly important over time. Allowing a field to be optional effectively _increases_ the API surface area and is a backward compatible change. So in theory, all new metric fields should be required unless it's clear about what the optionality means.

We've done a good job with this in the rest of the PR so I'm not really concerned about this particular metric. Just something to think about.

Reply:
Here I'm using the service type field to represent "where did someone initiate a gesture to view a cloudwatch logs entity from?" - many of those gestures will be embedded within service related views, but the primary entry point in the AWS Explorer will not be service specific. 

Some alternative ideas:
- use an open-ended string field (which is inherently optional), but accept that different clients may place different values in the field
- define an all new enum type like "cloudwatchGestureSource", which we can add specific gesture points to
- (other suggestions?)

Reply:
I've been leaning towards a single, open-ended `source` field for things like this though nothing is really set-in-stone here. It's less useful for comparing across clients, but really this level of detail is already difficult to compare anyway as we're quickly approaching into UI.

CWL is a special case because it does relate to many, many services. So having a dedicated field just to label which service the gesture relates to could work in combination with a generic `source` that differentiates between UI.

Like I want to know _where_ the user started the action and for _what_ purpose.

Reply:
Ok, how about we leave serviceType (and keep it optional), and add a open ended `source` field. That way we can represent the two axis (and account for gestures from AWS Explorer). I think we would be more interested in comparing service type across clients than the actual source, so this gives us a nice level of flexibility.